app/views/queue_items/index.html.haml

```diff
@@ -14,18 +14,21 @@
               %th(width="15%") Genre
               %th(width="15%") Remove
           %tbody
+          = form_tag my_queue_path, method: :patch, id: 'queue_items_form' do
             - @queue_items.each do |queue_item|
               %tr
                 %td
-                  %input.form-control(type="text" value="#{queue_item.position}")
+                  = text_field_tag "queue_ids_with_positions[#{queue_item.id}]", queue_item.position, class: 'form-control', form: 'queue_items_form'
```

Comment:
You dont need to pass exact `id` to `"queue_ids_with_positions[#{queue_item.id}]"` 

Just `text_field_tag "queue_items[][position]"` would be enough - like...
`%td= text_field_tag "queue_items[][position]", queue_item.position, class: "form-control"`